GoogleForCreators/web-stories-wp
# Dashboard: get author names via _embedded data to avoid excess call for users

## Summary
Gets story author names from `web-stories/v1/web-story` by utilizing WordPress `embed` parameter to eliminate a call to get all users so we can map the names to the story author id, which used to be the only piece of author info we had in the response of web stories. 

## Relevant Technical Choices
- Get `_embedded` key back in body response of web stories, use that to get author name. 
- Opted to specify we want to get embedded author data only, per this documentation:  [WP global parameters embed](https://developer.wordpress.org/rest-api/using-the-rest-api/global-parameters/#_embed)

## To-do

- [x] I'd like some eyes on the PHP change I made (`$response = rest_get_server()->envelope_response( $response, true );` to get the envelope response to return embedded results before swapping this from draft to review. 

## User-facing changes
- Should no longer see a call to `wp-json/wp/v2/users?per_page=100&_locale=user` if you open your dev tools -> network tab on dashboard load 

## Testing Instructions
- Verify the above request is no longer made
- Verify that author names are loading for my stories 
- Verify that karma and unit tests still run 

---



Addresses #4753
